pkg/cluster/install.go

```diff
@@ -57,109 +94,73 @@ func (m *manager) adminUpdate() []steps.Step {
 		// struct, so we'll rebuild the fpAuthorizer and use the error catching
 		// to advance
 		steps.AuthorizationRetryingAction(m.fpAuthorizer, m.fixupClusterSPObjectID),
-		steps.Action(m.fixInfraID), // Old clusters lacks infraID in the database. Which makes code prone to errors.
+		steps.Action(m.startVMs),
+		steps.Condition(m.apiServersReady, 30*time.Minute, true),
 	}
 
-	if isEverything {
-		toRun = append(toRun,
-			steps.Action(m.ensureResourceGroup), // re-create RP RBAC if needed after tenant migration
-			steps.Action(m.createOrUpdateDenyAssignment),
-			steps.Action(m.ensureServiceEndpoints),
-			steps.Action(m.populateRegistryStorageAccountName), // must go before migrateStorageAccounts
-			steps.Action(m.migrateStorageAccounts),
-			steps.Action(m.fixSSH),
-			// steps.Action(m.removePrivateDNSZone), // TODO(mj): re-enable once we communicate this out
-		)
+	// Generic fix-up actions that are fairly safe to always take, and don't require a running cluster
+	step0 := []steps.Step{
+		steps.Action(m.fixInfraID), // Old clusters lacks infraID in the database. Which makes code prone to errors.
 	}
 
-	if isEverything || isRenewCerts {
-		toRun = append(toRun,
-			steps.Action(m.populateDatabaseIntIP),
-		)
-	}
+	return append(bootstrap, step0...)
+}
 
-	// Make sure the VMs are switched on and we have an APIServer
-	toRun = append(toRun,
-		steps.Action(m.startVMs),
-		steps.Condition(m.apiServersReady, 30*time.Minute, true),
-	)
+func (m *manager) getGeneralFixesSteps() []steps.Step {
+	return []steps.Step{
+		steps.Action(m.ensureResourceGroup), // re-create RP RBAC if needed after tenant migration
+		steps.Action(m.createOrUpdateDenyAssignment),
+		steps.Action(m.ensureServiceEndpoints),
+		steps.Action(m.populateRegistryStorageAccountName), // must go before migrateStorageAccounts
+		steps.Action(m.migrateStorageAccounts),
+		steps.Action(m.fixSSH),
+		// steps.Action(m.removePrivateDNSZone), // TODO(mj): re-enable once we communicate this out
 
-	// Requires Kubernetes clients
-	if isEverything {
-		toRun = append(toRun,
-			steps.Action(m.fixSREKubeconfig),
-			steps.Action(m.fixUserAdminKubeconfig),
-			steps.Action(m.createOrUpdateRouterIPFromCluster),
-		)
-	}
+		steps.Action(m.fixSREKubeconfig),
+		steps.Action(m.fixUserAdminKubeconfig),
+		steps.Action(m.createOrUpdateRouterIPFromCluster),
```

Comment:
The action`populateDatabaseIntIP` that populates the internal LB IP for the API into the DB was run before these 3 actions in the old code. I wonder if there was any requirement for that, since these 3 steps will require the API to be up.

Same wonder with the `apiServersReady` condition after starting the VMs which is now in `getZerothSteps` function before the DB IP fix runs, but used to be after.

Reply:
The reason I moved this down is because `populateDatabaseIntIP` patches  `doc.OpenShiftCluster.Properties.APIServerProfile.IntIP`. This property in only 2 other tasks: 
- [fixMCSCert](https://github.com/mociarain/ARO-RP/blob/60758295008fbd905dab0b3e485996d8e4024a9b/pkg/cluster/fixmcscert.go#L27)
- [fixMCSUserData](https://github.com/mociarain/ARO-RP/blob/60758295008fbd905dab0b3e485996d8e4024a9b/pkg/cluster/fixmcsuserdata.go#L176)


For the `startVMs` task I moved that back into bootstrapping because it seems to fit there better conceptually and I'm assuming the tasks that come before it in the if blocks can't be pre-requisites because we start the VMs in each path but only run some tasks before we start the VMs in `isEverything`.